src/main/java/net/minestom/server/permission/PermissionHandler.java

```diff
@@ -100,13 +113,21 @@ default Permission getPermission(@NotNull String permissionName) {
      * @return true if the handler has the permission, false otherwise
      */
     default boolean hasPermission(@NotNull String permissionName, @Nullable PermissionVerifier permissionVerifier) {
-        final Permission permission = getPermission(permissionName);
+        boolean result = false;
 
+        final Permission permission = getPermission(permissionName);
         if (permission != null) {
             // Verify using the permission verifier
-            return permissionVerifier == null || permissionVerifier.isValid(permission.getNBTData());
+            result = permissionVerifier == null || permissionVerifier.isValid(permission.getNBTData());
+        }
+
+        if (this instanceof Player player) {
```

Comment:
I dont think the *exact same event* should override both the hasPermission method without a verifier and the one with a verifier. Possibly split it up to 2 events, or a nullable PermissionVerifier in the original event?
Again, just return the PermissionHandler here too.